### PR TITLE
Add contract bytecode/state dump + disassembly to yahcli

### PR DESCRIPTION
**Pull request by david-bakin-sl**
Add to `yahcli` commands for handling contracts:
* From a signed state file extract in a readable (text) format:
   * all contracts
   * all contracts' state (persisted storage)
* Given a contract's bytecode disassemble it
   * and perform some "enhanced" disassembly

There's usage documentation in a markdown file in there somewhere.  That's still be updated/improved.

(More contract bytecode analysis is in progress but not yet ready to review.)

**Related issue(s)**:

Fixes # TBD

** Notes for reviewers**:

This is a big pile of code. It works great - I've been using it to investigate contracts for various gnarly investigations for a couple of months now.  Review at your leisure.  It's a tool for use at the desktop, not a production service.  That's not by any means an ask that you be lenient on this review, just letting you know so you can set your priorities.

P.S. This thing really should be a plugin for the `pcli` tool.  Unfortunately, that isn't available now because `pcli` lives in the `platform-sdk` part of the repo which isn't, yet, fully integrated into the build system with the services part of the repo.  It can't live _under_ `platform-sdk` because it needs domain information from services (e.g., accounts, contract store, etc.) and the dependencies don't run in that direction.  But it also can't - at the moment! - live as a `pcli` plugin in the services part of the repo because the services part doesn't import the `swirlds-cli.jar` artifact.  We'll get there, though.

**Checklist**

- [coming] Documented (Code comments, README, etc.)
- [manual] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/test-clients/src/main/java/com/hedera/services/yahcli/commands/contract/00-TODO.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+* selectors
```
Comment (Nana-EC) [inline hedera-node/test-clients/src/main/java/com/hedera/services/yahcli/commands/contract/00-TODO.md:2]: I don't we should be checking in to-do files.
If there's a top level README you can not the product EPIC ticket there.
In that EPIC issue you can have the list of issues/features and check them off over time

File: hedera-node/test-clients/src/main/java/com/hedera/services/yahcli/commands/contract/00-running-yahcli-to-get-contract-info.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+id by looking in the file store using a (virtual blob) key which is a
+pair of the account id and `Type.CONTRACT_BYTECODE`.  The account id
+must be of a smart contract account.
+
+## Subcommands of `contract`
+
+The subcommands of the `contract` command are:
+
+* `yahcli contract dumprawcontracts` - dumps the bytecodes for all contracts in the signed state
+    - optionally eliminate duplicates
```
Comment (Nana-EC) [inline hedera-node/test-clients/src/main/java/com/hedera/services/yahcli/commands/contract/00-running-yahcli-to-get-contract-info.md:41]: Q: how? does this need a separate flag?

File: hedera-node/test-clients/src/main/java/com/hedera/services/yahcli/commands/contract/00-running-yahcli-to-get-contract-info.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+./yahcli contract dumprawcontracts --unique --with-ids --prefix=">" \
+   -f ~/Downloads/contract_state_hip_583_analysis/SignedState.swh | \
+   grep -E '^>' | cut -f 2,4 > raw-unique-contract-bytecodes.txt
+```
+
+- 2023-03-16: found 768 unique contracts
+
+### find only solidity contracts
+
+- consider that the solidity prelude is
```
Comment (Nana-EC) [inline hedera-node/test-clients/src/main/java/com/hedera/services/yahcli/commands/contract/00-running-yahcli-to-get-contract-info.md:84]: Expand on this. Not very common knowledge so let readers know most solidity smart contract bytecode starts with and maps to this and what it means whiles you're at it

File: hedera-node/test-clients/src/main/java/com/hedera/services/yahcli/commands/contract/ContractCommand.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        description = "Dealing with contracts")
+public class ContractCommand implements Callable<Integer> {
+
+    @ParentCommand
+    Yahcli yahcli;
+
+    @Override
+    public Integer call() throws Exception {
+        throw new CommandLine.ParameterException(
+                yahcli.getSpec().commandLine(), "Please specify a contract subcommand!");
```
Comment (Nana-EC) [inline hedera-node/test-clients/src/main/java/com/hedera/services/yahcli/commands/contract/ContractCommand.java:51]: Is there an easy was to provide the options or would it face becoming out of date or being too long?

File: hedera-node/test-clients/src/main/java/com/hedera/services/yahcli/commands/contract/evminfo/Assembly.java [hunk 5]
```diff
@@ -0,0 +1,565 @@
...
+            sb.setLength(goalColumn);
+        }
+
+        /**
+         * Check if this Line is a specific instruction opcodes
+         *
+         * <p>Convenience method for check to see if this line is an instruction with the given
+         * opcode.
+         */
+        default boolean thisLineIsA(final int opcode) {
```
Comment (Nana-EC) [inline hedera-node/test-clients/src/main/java/com/hedera/services/yahcli/commands/contract/evminfo/Assembly.java:90]: Why the overload of `thisLineIsA`?
Would this be better names `isCodeLineOpcodeMatch()` and `isCodeLineMneumonicMatch()`

File: hedera-node/test-clients/src/main/java/com/hedera/services/yahcli/commands/contract/evminfo/CodeLine.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    @Override
+    public void formatLine(@NonNull StringBuilder sb) {
+        if (Variant.DISPLAY_CODE_OFFSET.getValue()) {
+            extendWithBlanksTo(sb, Columns.CODE_OFFSET.getColumn());
+            sb.append("%5X".formatted(codeOffset));
+        }
+
+        if (Variant.DISPLAY_OPCODE_HEX.getValue()) {
+            extendWithBlanksTo(sb, Columns.OPCODE.getColumn());
+            sb.append("  %02X".formatted(opcode()));
```
Comment (Nana-EC) [inline hedera-node/test-clients/src/main/java/com/hedera/services/yahcli/commands/contract/evminfo/CodeLine.java:54]: Unknown spaces throw me off. Why this number of characters and why for this variant?

File: hedera-node/test-clients/src/main/java/com/hedera/services/yahcli/commands/contract/evminfo/CodeLine.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+        sb.append(Opcodes.getOpcode(opcode()).mnemonic());
+        if (0 != operandBytes.length) {
+            extendWithBlanksTo(sb, Columns.OPERAND.getColumn());
+            sb.append(hexer().formatHex(operandBytes));
+        }
+        if (!eolComment.isEmpty()) {
+            if (Columns.EOL_COMMENT.getColumn() - 1 > sb.length()) {
+                extendWithBlanksTo(sb, Columns.EOL_COMMENT.getColumn());
+            } else {
+                sb.append("  ");
```
Comment (Nana-EC) [inline hedera-node/test-clients/src/main/java/com/hedera/services/yahcli/commands/contract/evminfo/CodeLine.java:67]: Why this number of characters blanks and why here?

File: hedera-node/test-clients/src/main/java/com/hedera/services/yahcli/commands/contract/evminfo/DataPseudoOpLine.java [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+        operandBytes = null != operandBytes ? operandBytes : new byte[0];
+        eolComment = null != eolComment ? eolComment : "";
+    }
+
+    @Override
+    public @NonNull String mnemonic() {
+        return kind.name();
+    }
+
+    // NOTTODO: There's a LOT of commonality here with CodeLine! like maybe this should be a
```
Comment (Nana-EC) [inline hedera-node/test-clients/src/main/java/com/hedera/services/yahcli/commands/contract/evminfo/DataPseudoOpLine.java:50]: Remove the comment.

**Review by Nana-EC — COMMENTED** [review]
1st third thoughts

**Comment by david-bakin-sl** [comment]
Closed in favor of #7879

**Pull request closed without merging**